lib/spack/spack/test/cmd/buildcache.py

```diff
@@ -44,6 +44,69 @@ def mock_get_specs_multiarch(database, monkeypatch):
     )
 
 
+def tests_buildcache_copy(
+        tmpdir, mock_packages, mock_archive, mock_fetch, config,
+        install_mockery):
+
+    pkg = 'libdwarf'
+    install(pkg)
+    mirror_url = os.path.join(str(tmpdir), 'mirror_url')
+
+    buildcache('create', '-d', str(tmpdir), '--unsigned', pkg)
+
+    spec = Spec(pkg).concretized()
+    tarball = spack.binary_distribution.tarball_name(spec, '.spec.yaml')
+    tarball_path = spack.binary_distribution.tarball_path_name(spec, '.spack')
+
+    buildcache('copy', '--spec-yaml', os.path.join(str(tmpdir), 'build_cache',
+               tarball), '--base-dir', str(tmpdir),
+               '--destination-url', str(mirror_url))
+
+    assert os.path.exists(os.path.join(str(mirror_url),
+                          'build_cache', tarball_path))
+
+
+def tests_buildcache_save_yaml_root_spec(
+        tmpdir, mock_packages, mock_archive, mock_fetch, config,
+        install_mockery):
+    pkg = 'libdwarf'
+    install(pkg)
+    buildcache('create', '-d', str(tmpdir), '--unsigned', pkg)
+
+    spec = Spec(pkg).concretized()
+    tarball = spack.binary_distribution.tarball_name(spec, '.spec.yaml')
```

Comment:
The tests setup can be probably factored into a common function or fixture, since it looks very similar among the tests added in this file.

Reply:
Sorry for the late reply. This patch tests the following four commands. 
1. spack view statlink
2. spack buildcache copy
3. spack buildcache get-buildcache-name
4. spack buildcache save-yaml

Reply:
@fun2yang Apologies, I meant if you can comment in the test which operations within those commands you are testing and which invariants of the result you are asserting. Le me know if this requires any clarification.